### PR TITLE
KAFKA-17235 system test test_performance_service.py failed

### PR DESCRIPTION
**Pull request by brandboat**
related to https://issues.apache.org/jira/browse/KAFKA-17235

The root cause of this issue is a change we introduced in [KAFKA-16879](https://issues.apache.org/jira/browse/KAFKA-16879), where we modified the PushHttpMetricsReporter constructor to use Time.System [1]. However, Time.System doesn't exist in Kafka versions 0.8.2 and 0.9.

In test_performance_services.py, we have system tests for Kafka versions 0.8.2 and 0.9 [2]. These tests always use the tools JAR from the trunk branch, regardless of the Kafka version being tested [3], while the client JAR aligns with the Kafka version specified in the test suite [4]. This discrepancy is what causes the issue to arise.

To resolve this issue, we have a few options:
- Add Time.System to Kafka 0.8.2 and 0.9: This isn't practical, as we no longer maintain these versions.
~~- Modify the PushHttpMetricsReporter constructor to use new SystemTime() instead of Time.System: This would contradict the intent of [KAFKA-16879](https://issues.apache.org/jira/browse/KAFKA-16879), which aims to make SystemTime a singleton.~~
- Implement Time in PushHttpMetricsReporter use the time to get current time
- Remove system tests for Kafka 0.8.2 and 0.9 from test_performance_services.py

~~Given that we no longer maintain Kafka 0.8.2 and 0.9, and altering the constructor goes against the design goals of [KAFKA-16879](https://issues.apache.org/jira/browse/KAFKA-16879), option 3 appears to be the most feasible solution. However, I'm not sure whether it's acceptable to remove these old version tests. Maybe someone else has a better solution~~

"We'll proceed with option 2 since support for versions 0.8 and 0.9 is still required, meaning we can't remove those Kafka versions from the system tests."

[1] https://github.com/apache/kafka/blob/133f2b0f311ba1fd5a999f477bad38370c1772ca/tools/src/main/java/org/apache/kafka/tools/PushHttpMetricsReporter.java#L99
[2] https://github.com/apache/kafka/blob/0711d1558250c60807dfa815d46907eac8bb4b98/tests/kafkatest/sanity_checks/test_performance_services.py#L43-L44
[3] https://github.com/apache/kafka/blob/0711d1558250c60807dfa815d46907eac8bb4b98/tests/kafkatest/services/performance/producer_performance.py#L89
[4] https://github.com/apache/kafka/blob/0711d1558250c60807dfa815d46907eac8bb4b98/tests/kafkatest/services/performance/producer_performance.py#L80

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@brandboat nice finding!

The total supports of 0.8 and 0.9 are not dropped. At least, we expect the producer should work. Hence, I prefer to solution 2 - make `PushHttpMetricsReporter` work with 0.8

`PushHttpMetricsReporter` use the `time` to get current time, and hence we can do a bit refactor for it. For example: `PushHttpMetricsReporter` can implement the `Time` to offer the current time.

**Comment by brandboat** [comment]
> The total supports of 0.8 and 0.9 are not dropped. At least, we expect the producer should work. Hence, I prefer to solution 2 - make PushHttpMetricsReporter work with 0.8
>
> PushHttpMetricsReporter use the time to get current time, and hence we can do a bit refactor for it. For example: PushHttpMetricsReporter can implement the Time to offer the current time.

Thanks for the suggestion, I've addressed everything you mentioned, could you take another look ?

File: tools/src/main/java/org/apache/kafka/tools/PushHttpMetricsReporter.java [hunk 1]
```diff
@@ -96,7 +97,31 @@ public class PushHttpMetricsReporter implements MetricsReporter {
                             "producer/consumer/streams/connect instance");
 
     public PushHttpMetricsReporter() {
-        time = Time.SYSTEM;
+        // This is for backward compatibility with older versions like 0.8 and 0.9,
```
Comment (chia7712) [inline tools/src/main/java/org/apache/kafka/tools/PushHttpMetricsReporter.java]: Could you please includes following items in docs:

1. this is used by xxx E2E
2. the E2E run this report with older client jars
3. hence this class must be compatible with older client jars.

**Review by chia7712 — COMMENTED** [review]
@brandboat thanks for this fix. BTW, another fix is that we can change `time` from `Time` to `Supplier<Long>` to offer current time. WDYT?

File: tools/src/main/java/org/apache/kafka/tools/PushHttpMetricsReporter.java (hunk 1 above)
Comment (brandboat) [reply tools/src/main/java/org/apache/kafka/tools/PushHttpMetricsReporter.java]: > another fix is that we can change time from Time to Supplier<Long> to offer current time. WDYT?

Thanks for the review, this one looks better.

**Comment by chia7712** [comment]
@brandboat Could you please verify the e2e on your local?

**Comment by brandboat** [comment]
> @brandboat Could you please verify the e2e on your local?

Yes, I run cmd as follows
```shell
./gradlew clean build -x test
docker rm -f ducker01 ducker02 ducker03 ducker04 ducker05 ducker06 ducker07 ducker08 ducker09 ducker10 ducker11 ducker12 ducker13 ducker14
TC_PATHS="tests/kafkatest/sanity_checks/test_performance_services.py" bash tests/docker/run_tests.sh
```

and test passed

```shell
================================================================================
SESSION REPORT (ALL TESTS)
ducktape version: 0.11.4
session_id:       2024-08-05--002
run time:         3 minutes 13.730 seconds
tests run:        7
passed:           7
flaky:            0
failed:           0
ignored:          0
================================================================================
```

**Review by chia7712 — APPROVED** [review]
@brandboat thanks for the sharing. LGTM

I will backport this fix to 3.9 to stabilize E2E test. FYI @cmccabe

**Pull request merged**